cmd/buck/cli/init.go

```diff
@@ -40,113 +28,43 @@ Use the '--existing' flag to initialize from an existing remote bucket.
 Use the '--cid' flag to initialize from an existing UnixFS DAG.
 `,
 	Args: cobra.ExactArgs(0),
-	PreRun: func(c *cobra.Command, args []string) {
-		cmd.ExpandConfigVars(config.Viper, config.Flags)
-	},
 	Run: func(c *cobra.Command, args []string) {
-		root, err := os.Getwd()
-		if err != nil {
-			cmd.Fatal(err)
-		}
-		dir := filepath.Join(root, config.Dir)
-		if err = os.MkdirAll(dir, os.ModePerm); err != nil {
-			cmd.Fatal(err)
-		}
-		filename := filepath.Join(dir, config.Name+".yml")
-		if _, err := os.Stat(filename); err == nil {
-			cmd.Fatal(fmt.Errorf("bucket %s is already initialized", root))
-		}
+		conf, err := bucks.NewConfigFromCmd(c, ".")
```

Comment:
The normal way of creating a bucket is to supply a config that contains a path, key, and thread. But if the user wants to connect a command's flags / env vars to that config, they can use this.